### PR TITLE
mikutter: new package

### PR DESCRIPTION
**Pull request by midchildan**
###### Motivation for this change
mikutter is an extensible Twitter client.

###### Things done

- [ ] Tested using sandboxing
  ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS,
    or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file)
    on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] Linux
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

**Comment by AndersonTorres** [comment]
Two things:

1 - Put every logical change in a separate commit. As I have noted, you
modified pthread-stubs too; it deserves a commit for himself alone

2 - The format for new packages/expressions is "<name of software>: init at
<version>", and for updates is "<name of software>: <old version> -> <new
version>".




2017-05-17 7:11 GMT-03:00 midchildan <notifications@github.com>:

**Comment by midchildan** [comment]
Thanks. I have updated the commits.

**Comment by midchildan** [comment]
mikutter wasn't working well in a test environment, so I pushed a little fix that sets `GI_TYPELIB_PATH` when launching mikutter. It seems to be working fine now.

File: pkgs/development/libraries/pthread-stubs/default.nix [hunk 1]
```diff
@@ -0,0 +1,17 @@
+{stdenv, fetchurl}:
+
+stdenv.mkDerivation rec {
+  name = "pthread-stubs-${version}";
```
Comment (Mic92) [inline pkgs/development/libraries/pthread-stubs/default.nix]: we already have this package in `pkgs/servers/x11/xorg/default.nix`. It is reachable via `xlibs.libpthreadstubs`

File: pkgs/applications/networking/instant-messengers/mikutter/default.nix [hunk 2]
```diff
@@ -48,17 +32,27 @@ stdenv.mkDerivation rec {
...
     cp -rv . $out
-    makeWrapper $out/mikutter.rb $out/bin/mikutter \
-      --prefix PATH : "${ruby}/bin:${alsaUtils}/bin:${libnotify}/bin" \
-      --prefix GEM_HOME : "${env}/${env.ruby.gemPath}" \
-      --prefix GI_TYPELIB_PATH : "$GI_TYPELIB_PATH" \
+    mkdir $out/bin/
+    # hack wrapGAppsHook wants a file not a symlink
+    mv $out/mikutter.rb $out/bin/mikutter
+
+    gappsWrapperArgs+=(
```
Comment (Mic92) [inline pkgs/applications/networking/instant-messengers/mikutter/default.nix]: Using `wrapGAppsHook` is always a good idea.

File: pkgs/applications/networking/instant-messengers/mikutter/Gemfile [hunk 3]
```diff
@@ -0,0 +1,37 @@
+alias __source_distinct__ source
```
Comment (Mic92) [inline pkgs/applications/networking/instant-messengers/mikutter/Gemfile]: That file would be smaller if mikutter would had a gemspec.

File: pkgs/applications/networking/instant-messengers/mikutter/shell.nix [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+  buildInputs = with pkgs; [
+    bundix
+    bundler
+    pkgconfig
+    ruby
+    gdk_pixbuf
+    gtk2
+    xorg.libXdmcp
+    pcre
+    pthread_stubs
```
Comment (Mic92) [inline pkgs/applications/networking/instant-messengers/mikutter/shell.nix]: I think this file needs to be revisited now, as I have removed `pthread_stubs`.

File: pkgs/applications/networking/instant-messengers/mikutter/Gemfile (hunk 3 above)
Comment (midchildan) [inline pkgs/applications/networking/instant-messengers/mikutter/Gemfile]: Unfortunately, mikutter doesn't provide one.

File: pkgs/applications/networking/instant-messengers/mikutter/default.nix [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+    mkdir -p $out/share/mikutter $out/share/applications
+    ln -sv $out/core/skin $out/share/mikutter/skin
+    substituteAll ${./mikutter.desktop} $out/share/applications/mikutter.desktop
+  '';
+
+  meta = with stdenv.lib; {
+    description = "An extensible Twitter client";
+    homepage = "https://mikutter.hachune.net";
+    platforms = ruby.meta.platforms;
+    license = licenses.mit;
```
Comment (Mic92) [inline pkgs/applications/networking/instant-messengers/mikutter/default.nix]: Do you want to maintain this package?

Comment (midchildan) [inline pkgs/applications/networking/instant-messengers/mikutter/default.nix]: Sure :-)

Comment (Mic92) [inline pkgs/applications/networking/instant-messengers/mikutter/default.nix]: Then please add yourself as described here: https://github.com/nixos-users/wiki/wiki/Get-Involved#becoming-a-nixpkgs-maintainer

Comment (midchildan) [inline pkgs/applications/networking/instant-messengers/mikutter/default.nix]: Done

**Pull request closed without merging**